src/context.rs

```diff
@@ -106,9 +106,37 @@ fn test_should_add_a_service() {
     }
 
     let service = ServiceStub;
-    let mut foo = Context::new(false, Some("localhost".to_owned()), None, None);
+    let context = Context::shared(false, Some("localhost".to_owned()), None, None);
+    let mut foo = context.lock().unwrap();
 
     assert_eq!(foo.services.is_empty(), true);
     foo.add_service(Box::new(service));
     assert_eq!(foo.services.is_empty(), false);
+    assert_eq!(foo.services_count(), 1);
+
+    // Verify that the service with id "1" is there.
+    {
+        let service1 = foo.get_service("1");
+        match service1 {
+            Some(svc) => {
+                assert_eq!(svc.get_properties().id, "1");
+            }
+            None => assert!(false, "No service with id 1")
+        }
+    }
+
+    let id = "1".to_owned();
+    assert_eq!(foo.get_http_root_for_service(id.clone()),
+               "http://localhost:3000/services/1/");
+    assert_eq!(foo.get_ws_root_for_service(id.clone()),
+               "ws://localhost:4000/services/1/");
+
+    match foo.services_as_json() {
+        Ok(txt) => assert_eq!(txt, "{\"1\":{\"id\":\"1\",\"name\":\"dummy service\",\"description\":\"really nothing to see\",\"http_url\":\"2\",\"ws_url\":\"3\"}}"),
+        Err(err) => assert!(false, err)
```

Comment:
Actually, using `panic!()` doens't sound that great. If we use `#[should_panic]`, we can hide another error, which wasn't the one we expected. In this case, the test becomes a false positive, which can be harmful.